Sources/ReadingListKit/ReadingListModel.swift

```diff
@@ -137,6 +140,75 @@ public class ReadingListModel: ReadingListModelDelegate, ObservableObject {
         return ReadingListCellViewModel(id: id, title: title, subtitle: subtitle ?? "", contentURL: contentURL, thumbnailURL: thumbnailURL)
     }
 
+    // MARK: - Delegate Methods
+
+    func didFetchReadingListItems(urlStrings: [String], totalItemCount: Int) {
+        totalNumberOfItemsInReadingList = totalItemCount
+
+        displayMode = .normal
+        print("Set displayMode to: \(displayMode)")
+
+        if totalItemCount == 0 {
+            displayMode = .empty
+            print("Set displayMode to: \(displayMode)")
+        }
+
+        urlStrings.forEach { urlString in
+            Task {
+                guard let item = try? await pocketAccessLayer.getItemForURL(urlString) else { return }
+                let readingListItem = readingListItem(from: item)
+                await MainActor.run {
+                    readingListItems.append(readingListItem)
+                }
+            }
+        }
+    }
+
+    // MARK: - ReadingListItem Helpers
+
+    func readingListItem(from item: PocketGraph.ItemByURLQuery.Data.ItemByUrl) -> ReadingListCellViewModel {
+        let defaultImageURLString = "https://helios-i.mashable.com/imagery/articles/05fACELrEVc4kAfNQbhhcVh/hero-image.fill.size_1248x702.v1667556469.png"
+
+        let id = item.remoteID
+        let title = item.title ?? item.givenUrl
+        let subtitle = subtitle(for: item)
+        let contentURL = item.resolvedUrl ?? item.givenUrl
+        let thumbnailURL = image(for: item) ?? defaultImageURLString
+
+        return ReadingListCellViewModel(id: id, title: title, subtitle: subtitle, contentURL: contentURL, thumbnailURL: thumbnailURL)
+    }
+
+    func image(for item: PocketGraph.ItemByURLQuery.Data.ItemByUrl) -> String? {
+        item.syndicatedArticle?.mainImage ?? item.topImageUrl ?? item.domainMetadata?.logo
+    }
+
+    func subtitle(for item: PocketGraph.ItemByURLQuery.Data.ItemByUrl) -> String {
+        let host = item.domainMetadata?.name ?? host(from: item.givenUrl)
+
+        guard let host = host else {
+            return ""
+        }
+
+        guard let timeToRead = item.timeToRead else { return host }
+
+        return host + " • " + String(describing: timeToRead) + " min"
+    }
+
+    func host(from url: String) -> String? {
+        guard let url = URL(string: url) else { return nil }
+        return url.host
+    }
+
+    // MARK: - Notification Events
+
+    @objc
+    func userAuthDidChange() {
+        readingListItems.removeAll()
+        displayMode = .normal
+        paginationCursor = nil
+        fetchMoreReadingList()
```

Comment:
Wondering if it's worth the minor addition of only fetching more items if the user _is_ logged in. Otherwise, we're guaranteeing an apollo request to fail because there are no credentials.

Reply:
That would mean the model needs to know and care about auth states.
Since Apollo already does this and throws us back an error I figured that was the way to go.

Reply:
Hmm… it's starting to feel like the API client is the basis for whether a user is logged in our out, based on the result of a network call. So, when the client fails because the user is not authenticated, that makes its way back to the model, here, which then updates itself causing a re-render for a "logged out" view. This goes hand-in-hand with another comment of mine re: sending an object with a notification.

Reply:
Yes, it's starting to seem like a solution.